command/ssh.go

```diff
@@ -0,0 +1,180 @@
+		err = ioutil.WriteFile(sshDynamicKeyFileName, []byte(sshDynamicKey), 0600)
+		sshCmdArgs = append(sshCmdArgs, []string{"-i", sshDynamicKeyFileName}...)
+
+	} else if keySecret.Data["key_type"].(string) == ssh.KeyTypeOTP {
+		fmt.Printf("OTP for the session is %s\n", string(keySecret.Data["key"].(string)))
+	} else {
+		// Intentionally not mentioning the exact error
+		c.Ui.Error("Error creating key")
+	}
+	sshCmdArgs = append(sshCmdArgs, []string{"-p", port}...)
+	sshCmdArgs = append(sshCmdArgs, args...)
+
+	sshCmd := exec.Command("ssh", sshCmdArgs...)
+	sshCmd.Stdin = os.Stdin
+	sshCmd.Stdout = os.Stdout
+
+	err = sshCmd.Run()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error while running ssh command:%s", err))
+	}
+
+	if keySecret.Data["key_type"].(string) == ssh.KeyTypeDynamic {
+		err = os.Remove(sshDynamicKeyFileName)
+		if err != nil {
+			// Intentionally not mentioning the exact error
+			c.Ui.Error("Error cleaning up")
+		}
+	}
+
+	err = client.SSH().KeyRevoke(keySecret.LeaseID)
+	if err != nil {
+		// Intentionally not mentioning the exact error
+		c.Ui.Error("Error cleaning up")
+	}
+
+	return 0
+}
+
+func setDefaultRole(client *api.Client, ip string) (string, error) {
+	data := map[string]interface{}{
+		"ip": ip,
+	}
+	secret, err := client.Logical().Write("ssh/lookup", data)
+	if err != nil {
+		return "", fmt.Errorf("Error finding roles for IP '%s':%s", ip, err)
+
+	}
+	if secret == nil {
+		return "", fmt.Errorf("Error finding roles for IP '%s':%s", ip, err)
+	}
+
+	if secret.Data["roles"] == nil {
+		return "", fmt.Errorf("IP '%s' not registered under any role", ip)
+	}
+
+	if len(secret.Data["roles"].([]interface{})) == 1 {
+		return secret.Data["roles"].([]interface{})[0].(string), nil
+	} else {
+		return "", fmt.Errorf("Multiple roles for IP '%s'. Select one of '%s' using '-role' option", ip, secret.Data["roles"])
+	}
+}
+
+func (c *SSHCommand) Synopsis() string {
+	return "Initiate a SSH session"
+}
+
+func (c *SSHCommand) Help() string {
+	helpText := `
+Usage: vault ssh [options] username@ip
+
+  Establishes an SSH connection with the target machine.
+
+  This command generates a key and uses it to establish an SSH
+  connection with the target machine. This operation requires
+  that SSH backend is mounted and at least one 'role' be registed
+  with vault at priori.
+
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+SSH Options:
+
+  -role                 Mention the role to be used to create dynamic key.
+  			Each IP is associated with a role. To see the associated
+			roles with IP, use "lookup" endpoint. If you are certain that
+			there is only one role associated with the IP, you can
+			skip mentioning the role. It will be chosen by default.
+			If there are no roless associated with the IP, register
+			the CIDR block of that IP using the "roles/" endpoint.
+
+  -port                 Port number to use for SSH connection. This defaults to port 22.
+`
```

Comment:
We need to support a few other options:
* `-no-exec` to just provide the username / password / key without actually invoking SSH (if you want to use rsync / scp or similar, or to customize how you invoke SSH)
* `-path` in case the SSH backend is mounted somewhere different like `ssh-prod/` and `ssh-stage/`